### PR TITLE
feat: Refactor WebSocket, JWT and PASETO Middleware to Refactor Filter to Next and Support Flexible Token Extractors

### PR DESCRIPTION
**Pull request by sixcolors**
This PR introduces a refactor to both the WebSocket, JWT and PASETO middleware packages, enabling flexible and extensible token extraction via a new `Extractor` abstraction. Key improvements and changes include:

### WebSocket Middleware
- **Replaces `Filter` with `Next`**, updated for consistency.

### JWT Middleware
- **Replaces `TokenLookup` and `AuthScheme`** with an `Extractor` field in config, supporting extraction from headers, cookies, query params, form data, or path params.
- **Adds built-in extractors:** `FromAuthHeader`, `FromHeader`, `FromCookie`, `FromQuery`, `FromForm`, `FromParam`, and `Chain` for fallback extraction.
- **Default extractor** uses `Authorization` header with `"Bearer"` scheme.
- **Improved error handling** for missing or malformed tokens via standardized `ErrMissingToken`.
- **Removes legacy lookup logic** for cleaner, easier-to-maintain code.
- **Adds comprehensive unit tests** for all extractors and extraction failure cases.
- **README updated** to document new extractor options and migration guidance.

### PASETO Middleware
- **Replaces `TokenLookup` and `TokenPrefix`** with an `Extractor` field in config for consistent extraction logic.
- **Provides same built-in extractors** as JWT for all supported sources and chaining.
- **Default extractor** uses `Authorization` header with `"Bearer"` scheme.
- **Test suite expanded** to cover extractor edge cases and chaining.
- **README updated** with new extractor usage, migration steps, and additional examples.

### Other Improvements
- **Consistent naming** (`Next` replaces `Filter`) across middleware packages.
- **Removes unused legacy helper logic** for token acquisition.
- **WebSocket middleware** updated for consistent config naming (`Next`).

---

**Migration Notes:**
- To migrate from `TokenLookup` or `TokenPrefix`, use the appropriate extractor, e.g.:
  ```go
  jwtware.New(jwtware.Config{
      Extractor: jwtware.FromAuthHeader("Bearer"),
  })
  pasetoware.New(pasetoware.Config{
      Extractor: pasetoware.FromAuthHeader("Bearer"),
  })
  ```
- See README for more examples, including extracting from cookies, query params, or chaining multiple extractors.

## Summary by CodeRabbit

* **New Features**
  * Unified extractor-based token retrieval (headers, cookies, query, params, form, chainable) with helpers for auth-header and cookie extraction; token pre-processing hook, multi-key signing and JWK Set URL support, exported signing-key and customizable key function, and improved websocket recovery handler.

* **Bug Fixes**
  * Clearer missing-token errors, safer context token access, and consistent middleware skip behavior; default success path now continues middleware.

* **Refactor**
  * Config option renamed for clarity; token-location fields replaced by extractor-based configuration.

* **Documentation**
  * Added Available Extractors, security guidance, and migration examples.

### CODE REVIEW

File: jwt/README.md [hunk 1]
```diff
@@ -38,13 +38,16 @@ jwtware.FromContext(c fiber.Ctx) *jwt.Token
...
-| TokenLookup    | `string`                        | TokenLookup is a string in the form of `<source>:<name>` that is used                                                                                   | `
+| SigningKey     | `SigningKey`                     | Signing key to validate token. Used as fallback if SigningKeys is empty.                                                                            | `nil`                        |
+| SigningKeys    | `map[string]SigningKey`         | Map of signing keys to validate token with kid field usage.                                                                                             | `nil`                        |
+| Claims         | `jwt.Claim`                     | Claims are extendable claims data defining token content.                                                                                               | `jwt.MapClaims{}`            |
+| Extractor      | `Extractor`                     | Extractor defines a function to extract the token from the request.                                                                                     | `FromAuthHeader("Authorization", "Bearer")` |
```
[the remark this answers is not in the document]
Comment (sixcolors) [reply jwt/README.md]: @gemini-code-assist make a comittable suggestion to fix

File: websocket/websocket.go [hunk 2]
```diff
@@ -131,19 +131,22 @@ func New(handler func(*Conn), config ...Config) fiber.Handler {
...
+		}
 
 		// headers
-		c.RequestCtx().Request.Header.VisitAll(func(key, value []byte) {
-			conn.headers[string(key)] = string(value)
-		})
+		headers := c.RequestCtx().Request.Header.All()
+		for key, values := range headers {
+			conn.headers[string(key)] = string(values)
+		}
```
[the remark this answers is not in the document]
Comment (sixcolors) [reply websocket/websocket.go:149]: @gemini-code-assist

In fasthttp v1.65.0, the Args type (used for query arguments) includes:
 - All(): A new method returning an iterator over key-value pairs (replaces VisitAll for modern iteration).
 - VisitAll(): Deprecated method that internally uses All() for backward compatibility.
 - Other methods like Peek(), Set(), Add(), etc., for manipulating query args.
 This version supports the newer All() method, which wasn't available in earlier versions like v1.51.0.
 
 The go.mod is updated in the PR.

[the remark this answers is not in the document]
Comment (sixcolors) [reply websocket/websocket.go:149]: I think you are wrong `// All returns an iterator over key-value pairs in h.`

File: paseto/README.md [hunk 3]
```diff
@@ -246,32 +265,68 @@
...
+		Extractor:    pasetoware.FromQuery("token"),
+	}))
+
+	app.Get("/protected", func(c fiber.Ctx) error {
+		return c.SendString("Protected route")
+	})
+
+	app.Listen(":8080")
+}
 ```
```
[the remark this answers is not in the document]
Comment (ReneWerner87) [reply paseto/README.md:351]: @sixcolors

**Comment by sixcolors** [comment]
Wait on https://github.com/gofiber/fiber/pull/3725
